internal/relay/relayer.go

```diff
@@ -163,10 +160,13 @@ func (r *Relayer) processMessageTX(ctx context.Context, m *MessageTX) error {
 				// TODO: should we stop after a first error
 				return fmt.Errorf("failed to save last height of query: %w", err)
 			}
-			r.logger.Debug("block completely processed", zap.Uint64("query_id", m.QueryId), zap.Uint64("processed_height", lastProcessedHeight), zap.Uint64("next_height_to_process", tx.Height))
+			r.logger.Debug("block completely processed",
+				zap.Uint64("query_id", m.QueryId),
+				zap.Uint64("processed_height", lastProcessedHeight),
+				zap.Uint64("next_height_to_process", tx.Height))
 		}
 		lastProcessedHeight = tx.Height
-		err := r.txProcessor.ProcessAndSubmit(ctx, m.QueryId, tx)
+		err := r.txProcessor.ProcessAndSubmit(ctx, m.QueryId, tx, submittedTxsTasksQueue)
```

Comment:
don't really like this channel being passed from main to this point through the relayer whereas 1) the relayer barely uses it; 2) this channel looks like a parameter for the TxProcessor but we're not going to use any other channels here, so this chan can be set simply on initialisation. wyt about making it the following way?
1. instantiate SubmitChecker from the TxProcessor;
2. init a channel that will be only visible for the two above mentioned guys at the TxProcessor creation level;

as a result:
- simplified startup in the main func;
- simplified relayer (the channel removed, only simple `r.txProcessor.ProcessAndSubmit(ctx, m.QueryId, tx)` call here);
- more meaningful TxSubmitChecker on the TxProcessor level that uses it directly.

optionally, we can also address the comments [above](https://github.com/neutron-org/neutron-query-relayer/pull/44/files#r1002908192) and create a TxProcessor (and TxSubmitChecker) only when we really need them, i.e. when we have tx handling turned on

Reply:
`instantiate SubmitChecker from the TxProcessor` — no, all moving parts should be run inside the main function

Reply:
but i can consider adding the channel to tx processor during instantiation

Reply:
> instantiate SubmitChecker from the TxProcessor — no, all moving parts should be run inside the main function

why so? this is not the first time you insist on it while I still can't see the reason. why can't an entity that is very much related to one and only one other entity can't be run and shut down during the latter initialisation? we have a tx processor and tx submit checker here which take care of a single area: tx queries. why can't the whole this area be controlled by the tx processor that has the tx submit checker as one of its, so to say, parts?

Reply:
resolved in dms